tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Analyzer.scala

```diff
@@ -349,7 +350,80 @@ final class Analyzer(semantics: Semantics, outputMode: OutputMode,
           None
         }
       }
-      loop(this)
+      val existing =
+        if (isScalaClass) loop(this)
+        else methodInfos.get(methodName).filter(!_.isAbstract)
```

Comment:
It is so because, eventually, in JS, default methods cannot be made to inherit through any JS mechanism. Class methods are inherited through JS prototype chains, but interface methods cannot be inherited at the JS level.

If we chose to keep them inherited at the IR level, then desugaring would have to introduce indirections instead of the analyzer. But those indirections could not be removed by the optimizer like those produced by the analyzer are (through simple inlining). Not to mention `GenIncOptimizer` would be considerably more complicated.

In the JVM bytecode, they are obviously inherited (otherwise, default methods would exist at the bytecode level). That's why we allow them to be inherited before the analyzer (at the .sjsir level). The analyzer resolves the calls once and for all, creating statically linked bridges from there.